### PR TITLE
Return an expression instead of conditionally returning true/false

### PR DESCRIPTION
**Pull request by eugmes**
This replaces code that conditionally returns `true` or `false` after a test by more concise code that just returns the tested condition.

### CODE REVIEW

**Comment by jukkar** [comment]
At least I find it easier to read the original code even if it makes more lines. In fact I would actually change the first commit to use the
```
if (cond) {
     do something;
     return true;
}

return false;
```
as the else branch is not really needed and remove the second commit.

File: drivers/dma/dma_stm32_v1.c [hunk 1]
```diff
@@ -410,44 +410,21 @@ bool stm32_dma_check_fifo_mburst(LL_DMA_InitTypeDef *DMAx)
 		case LL_DMA_MBURST_INC4:
 			return true;
 		case LL_DMA_MBURST_INC8:
-			if (fifo_level == LL_DMA_FIFOTHRESHOLD_1_2 ||
-			    fifo_level == LL_DMA_FIFOTHRESHOLD_FULL) {
-				return true;
-			} else {
-				return false;
-			}
+			return fifo_level == LL_DMA_FIFOTHRESHOLD_1_2 || fifo_level == LL_DMA_FIFOTHRESHOLD_FULL;
```
Comment (tbursztyka) [inline drivers/dma/dma_stm32_v1.c:410]: how is this more readable? (and it's not fixing any issue so far)

Plus:
- you forgot the necessary () 
- and the lines are too long.

File: arch/arc/core/fault.c [hunk 2]
```diff
@@ -91,12 +91,8 @@ static bool z_check_thread_stack_fail(const uint32_t fault_addr, uint32_t sp)
 	  * overflow if the stack pointer is at or below the end of the guard
 	  * region.
 	  */
-	if (sp <= guard_end && fault_addr < guard_end &&
-	    fault_addr >= guard_start) {
-		return true;
-	}
-
-	return false;
+	return sp <= guard_end && fault_addr < guard_end &&
```
Comment (tbursztyka) [inline arch/arc/core/fault.c:94]: That's super ugly...

**Review by tbursztyka — CHANGES_REQUESTED** [review]

File: subsys/usb/usb_transfer.c [hunk 3]
```diff
@@ -61,11 +61,7 @@ bool usb_transfer_is_busy(uint8_t ep)
 {
 	struct usb_transfer_data *trans = usb_ep_get_transfer(ep);
 
-	if (trans && trans->status == -EBUSY) {
-		return true;
-	}
-
-	return false;
+	return trans && trans->status == -EBUSY;
```
Comment (jfischer-phytec-iot) [inline subsys/usb/usb_transfer.c:64]: I do not see the reason to change it. AFAIK in C the type of logical operators is integer and therefore the existing code you replace is more correct.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

**Pull request closed without merging**